### PR TITLE
NEP: Contract Basic Method Guideline

### PR DESCRIPTION
**Pull request by superboyiii**
Close https://github.com/neo-project/proposals/issues/135

### CODE REVIEW

**Comment by erikzhang** [comment]
According to NEP-1, maybe this can be an **Informational** NEP.

> An Informational NEP describes a NEO design issue, or provides general guidelines or information to the NEO community, but does not propose a new feature. Informational NEPs do not necessarily represent a NEO community consensus or recommendation, so users and implementors are free to ignore Informational NEPs or follow their advice.

**Review by shargon — COMMENTED** [review]
Mention `_deploy` method and the raised notification by the syscall?

**Comment by superboyiii** [comment]
> Mention `_deploy` method and the raised notification by the syscall?

Done.

File: nep-19.mediawiki [hunk 1]
```diff
@@ -0,0 +1,151 @@
+<pre>
+  NEP:
+  Title: Contract Basic Methods Standard
```
Comment (erikzhang) [inline nep-19.mediawiki:3]: `Contract Basic Methods Guideline`?

Comment (superboyiii) [reply nep-19.mediawiki]: > `Contract Basic Methods Guideline`?

Done.

**Comment by superboyiii** [comment]
@erikzhang @shargon Any other advice?

**Review by shargon — APPROVED** [review]

File: nep-19.mediawiki [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+<pre>
+{
+  "name": "destroy",
+  "safe": false,
+  "parameters": [],
+  "returntype": "Void"
+}
+</pre>
+This function can delete all the storage of this contract.
```
Comment (shargon) [inline nep-19.mediawiki:96]: Mention that the owner MUST send his tokens to other place?

Comment (superboyiii) [reply nep-19.mediawiki:96]: > Mention that the owner MUST send his tokens to other place?

Done

**Review by shargon — APPROVED** [review]

**Pull request closed without merging**